plugins/modules/win_dhcp_lease.py

```diff
@@ -0,0 +1,144 @@
+  - Task should be delegated to a Windows DHCP Server
+options:
+  type:
+    description:
+      - The type of DHCP address.
+      - Leases expire as defined by l(duration).
+      - When l(duration) is not specified, the server default is used.
+      - Reservations are permanent.
+    type: str
+    default: reservation
+    choices: [ reservation, lease ]
+  state:
+    description:
+      - Specifies the desired state of the DHCP lease or reservation.
+    type: str
+    default: present
+    choices: [ present, absent ]
+  ip:
+    description:
+      - The IPv4 address of the client server/computer.
+      - This is a required parameter, if l(mac) is not set.
+      - Can be used to identify an existing lease/reservation, instead of l(mac).
+    type: str
+    required: no
+  scope_id:
+    description:
+      - Specifies the scope identifier as defined by the DHCP server.
+      - This is a required parameter, if l(state=present) and the reservation or lease
+        doesn't already exist. Not required if updating an existing lease or reservation.
+    type: str
+  mac:
+    description:
+      - Specifies the client identifier to be set on the IPv4 address.
+      - This is a required parameter, if l(ip) is not set.
+      - Windows clients use the MAC address as the client ID.
+      - Linux and other operating systems can use other types of identifiers.
+      - Can be used to identify an existing lease/reservation, instead of l(ip).
+    type: str
+  duration:
+    description:
+      - Specifies the duration of the DHCP lease in days.
+      - The duration value only applies to l(type=lease).
+      - Defaults to the duration specified by the DHCP server
+        configuration.
+      - Only applicable to l(type=lease).
+    type: int
+  dns_hostname:
+    description:
+      - Specifies the DNS hostname of the client for which the IP address
+        lease is to be added.
+    type: str
+  dns_regtype:
+    description:
+      - Indicates the type of DNS record to be registered by the DHCP.
+        server service for this lease.
+      - l(a) results in an A record being registered.
+      - l(aptr) results in both A and PTR records to be registered.
+      - l(noreg) results in no DNS records being registered.
+    type: str
+    default: aptr
+    choices: [ aptr, a, noreg ]
+  reservation_name:
+    description:
+      - Specifies the name of the reservation being created.
+      - Only applicable to l(type=reservation).
+    type: str
+  description:
+    description:
+      - Specifies the description for reservation being created.
+      - Only applicable to l(type=reservation).
+    type: str
+'''
+
+EXAMPLES = r'''
+- name: Ensure DHCP reservation exists
+  win_dhcp_lease:
+    type: reservation
+    ip: 192.168.100.205
+    scope_id: 192.168.100.0
+    mac: 00b18ad15a1f
+    dns_hostname: "{{ ansible_inventory }}"
+    description: Testing Server
+  delegate_to: dhcp-raju-euc.vmware.com
```

Comment:
Is there a point to the `delegate_to` parts of these examples? It seems to be pretty specific to your environment.

Reply:
Just random examples to show the common use case (they are not real hostnames). They can be removed.